behave/features/steps/specialist_provider.py

```diff
@@ -366,5 +370,67 @@ def step_impl(context):
     assert CASE_SPLIT_TEXT in full_case_log, "Could not find split case in operators dashboard"
 
 
+@step(u'I am on the CSV upload page')
+def step_impl(context):
+    csv_page = f"{CLA_FRONTEND_URL}/provider/csvupload/"
+    context.helperfunc.open(csv_page)
+    page = f"/provider/csvupload/"
+    wait_until_page_is_loaded(page, context)
+    header = context.helperfunc.find_by_xpath("//header/h2[text()='CSV Upload']")
+    assert header is not None
+
+
+@step(u'I select \'Choose file\' and upload a csv file')
+def step_impl(context):
+    input_csv = context.helperfunc.find_by_xpath("//input[@name='csvfile']")
+    # test will fail if file can't be found or upload fails.
+    input_csv.send_keys(os.getcwd() + "/uploads/csvupload.csv")
+
 
+@step(u'I select the month and year for the uploaded csv file')
+def step_impl(context):
+    date_now = datetime.datetime.now().date()
+    # Alter date to be the first day of month
+    # Minis one month because current month will not be visible as option
+    new_date = date_now.replace(day=1, month=date_now.month - 1).strftime("%Y-%m-%d")
+    select = Select(context.helperfunc.find_by_xpath(f'//form[@name="csvForm"]/label/select'))
+    assert select is not None
+    try:
+        select.select_by_value(f"string:{new_date}")
+    except StaleElementReferenceException:
+        assert False, f"Could find date month or year in select options"
```

Comment:
error message incorrect, should be could not find. Also, is this the only cause of a stale element exception? Maybe consider including the date you are looking for in the error message so can check if it is actually wrong or is a flaky test?

Reply:
Changed and added a comment about checking the screenshot as well to make sure it's not the select option being stale.